### PR TITLE
chore!: reorganized stories into Atomic Design and TypeScript

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## Reorganized stories into Atomic Design and TypeScript

This is as absurdly large as it seems... I'd recommend using the deployed styleguide as your primary starting reference to review. 🙃 

Major changes:

* Use basic `Templating/` helpers to force a bit more standard formatting & organization
* Moved all stories to **Atom**, **Molecule**, or **Organism** per [Atomic Design](https://bradfrost.com/blog/post/atomic-web-design/)
    * **Colors** and **Layouts** are in a new **Foundations** section
* Fleshed out the root `About.stories.mdx` and added smaller equivalents as landing pages for ☝️ stories
* Merged `gamut-templates` back into `gamut`
* Converted stories to TypeScript _(it was super helpful in organizing them with `decoratedStory`)_
    * Also fixed a bunch of type errors and Storybook complaints from stories... (almost) no more annoying red text in `yarn start`

### Notes

Screenshot of the grid on the new landing page

I also made a few changes to components that I'll call out in comments and copy into their own PRs. They were generally components I already had some context into, such as improving the API in the not-yet-used ProgressBar to better match how the story should look.

For those looking on and wondering why this giant PR is sitting here: we'll be discussing it cross-discipline this week. 😇

### CODE REVIEW

File: packages/gamut/src/ProgressBar/index.tsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ export type ProgressBarProps = {
   /**
    * Whether to increase size and display the percentage as text.
    */
-  displayLabel?: boolean;
+  large?: boolean;
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/ProgressBar/index.tsx:12]: This will be split into its own PR.

Basically, the "large" name is a better moniker than "display label". I'm also going to have to add in an optional prop that allows changing what gets narrated, since progress bars might visually indicate a non-zero amount when at 0%.

File: packages/gamut/src/index.tsx [hunk 2]
```diff
@@ -7,26 +7,24 @@ export * from './Card/CardShell';
...
+export * from './Interstitial';
 export * from './Logo';
 export * from './Markdown';
 export * from './Menus/MenuItem';
 export * from './Menus/SideMenu';
-
-// Temporarily exporting under two names to ease any breaking changes
-// eslint-disable-next-line import/export
 export * from './Overlay';
-export { Overlay as Modal } from './Overlay';
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/index.tsx:22]: Also in #687.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.9-alpha.d7a82d.0+d7a82d2d<br/>@codecademy/gamut@2.15.1-alpha.d7a82d.4+d7a82d2d<br/></pre>

File: packages/gamut/src/Card/BannerCard.tsx [hunk 3]
```diff
@@ -1,28 +0,0 @@
-import React, { ReactNode } from 'react';
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Card/BannerCard.tsx:1]: `BannerCard` and `CoverCard` are unused and nobody spoke out to stop me. 🔪

File: packages/gamut/src/ContentContainer/index.tsx [hunk 4]
```diff
@@ -13,10 +13,10 @@ export type ContentContainerProps = {
 export const ContentContainer: React.FC<ContentContainerProps> = ({
   children,
   className,
-  el,
+  el = 'div',
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/ContentContainer/index.tsx:16]: Storybook recognizes both `defaultProps` and `=`s in destructuring as ways of declaring default prop values. It doesn't recognize the `|| 'div'` style below.

File: packages/gamut/src/ProgressBar/index.tsx [hunk 5]
```diff
@@ -22,17 +22,17 @@ export type ProgressBarProps = {
 export type ProgressBarStyle = {
   backgroundColor: string;
   barColor: string;
-  fontColor: string;
+  fontColor?: string;
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/ProgressBar/index.tsx:25]: Additionally, font color should be optional, because small progress bars don't need it.

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8b3f7ccc95e68054962b0d--codecademy-styleguide.netlify.com>https://5e8b3f7ccc95e68054962b0d--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8b3f7ccc95e68054962b0d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8b43dacc95e6f05196296c--codecademy-styleguide.netlify.com>https://5e8b43dacc95e6f05196296c--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8b43dacc95e6f05196296c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.9-alpha.08db1f.2+08db1f8a<br/>@codecademy/gamut@2.15.1-alpha.08db1f.6+08db1f8a<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8b46e8da95f4ab0963aa6c--codecademy-styleguide.netlify.com>https://5e8b46e8da95f4ab0963aa6c--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8b46e8da95f4ab0963aa6c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8b4902da95f40d4663a410--codecademy-styleguide.netlify.com>https://5e8b4902da95f40d4663a410--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8b4902da95f40d4663a410>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8b4e656ee6573f3b9fe0df--codecademy-styleguide.netlify.com>https://5e8b4e656ee6573f3b9fe0df--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8b4e656ee6573f3b9fe0df>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8b525681802ba20e9971b1--codecademy-styleguide.netlify.com>https://5e8b525681802ba20e9971b1--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8b525681802ba20e9971b1>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8b8585f55cf902166a2c8a--codecademy-styleguide.netlify.com>https://5e8b8585f55cf902166a2c8a--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8b8585f55cf902166a2c8a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8b8803a58458641e6caad5--codecademy-styleguide.netlify.com>https://5e8b8803a58458641e6caad5--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8b8803a58458641e6caad5>Deploy Logs</a></p>

File: codecov.yml [hunk 6]
```diff
@@ -6,5 +6,8 @@ codecov:
 
 comment:
   after_n_builds: 4 # Must be the # of parallel test builds in CircleCI
-  layout: "diff, files"
+  layout: 'diff, files'
   require_changes: true
+
+ignore:
+  - 'packages/styleguide'
```
Comment (JoshuaKGoldberg) [inline codecov.yml:13]: ...doesn't appear to be working... 🙁

Comment (jakemhiller) [reply codecov.yml:13]: might need to be merged into master to really take effect... i know it bases some stuff off of the state of master

Comment (JoshuaKGoldberg) [reply codecov.yml:13]: I can send a separate PR. I'm also curious whether we should put this in codecov.yml or in Jest.

Comment (JoshuaKGoldberg) [reply codecov.yml:13]: https://github.com/Codecademy/client-modules/pull/701

Comment (JoshuaKGoldberg) [reply codecov.yml:13]: Heh, Codecov corrected itself. Okie. Never mind this 🙂 🙃 .

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8bd036114e5866818e5b2e--codecademy-styleguide.netlify.com>https://5e8bd036114e5866818e5b2e--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8bd036114e5866818e5b2e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8d02bf821f505e7f543494--codecademy-styleguide.netlify.com>https://5e8d02bf821f505e7f543494--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8d02bf821f505e7f543494>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.9-alpha.a280c2.13+a280c233<br/>@codecademy/gamut@2.15.1-alpha.a280c2.17+a280c233<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8d0996fb63c1b7b537daed--codecademy-styleguide.netlify.com>https://5e8d0996fb63c1b7b537daed--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8d0996fb63c1b7b537daed>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.9-alpha.9bad4d.14+9bad4daa<br/>@codecademy/gamut@2.15.1-alpha.9bad4d.18+9bad4daa<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.9-alpha.549c4d.15+549c4d3d<br/>@codecademy/gamut@2.15.1-alpha.549c4d.19+549c4d3d<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.15-alpha.1506fd.0<br/>@codecademy/gamut@2.17.5-alpha.1506fd.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8d12d6d8db9c0203a94855--codecademy-styleguide.netlify.com>https://5e8d12d6d8db9c0203a94855--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8d12d6d8db9c0203a94855>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8d149244d1145857e93ac3--codecademy-styleguide.netlify.com>https://5e8d149244d1145857e93ac3--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8d149244d1145857e93ac3>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.15-alpha.bde1ce.0<br/>@codecademy/gamut@2.17.5-alpha.bde1ce.0<br/></pre>

File: packages/gamut/src/Layout/index.tsx [hunk 7]
```diff
@@ -1,2 +1,3 @@
-export { default as LayoutGrid } from './Grid';
-export { default as Column } from './Column';
+export * from './Column';
+export * from './LayoutGrid';
+export * from './types';
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Layout/index.tsx:3]: FYI @codecaaron

File: packages/gamut/src/Truncate/index.tsx [hunk 8]
```diff
@@ -3,16 +3,16 @@ import TruncateMarkup from 'react-truncate-markup';
...
   as?: 'div' | 'span' | 'p';
   /** number of the maximum lines to display, pass false to disable truncation */
-  lines: number | false;
+  lines?: number | false;
   /** Callback indicating if truncation was necessary */
   onTruncate?: (truncated: boolean) => void;
 };
 
-export const Truncate: React.FC<TruncateTextProps> = ({
+export const Truncate: React.FC<TruncateProps> = ({
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Truncate/index.tsx:15]: FYI @codecaaron

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.16-alpha.f3a2b6.0<br/>@codecademy/gamut@2.18.1-alpha.f3a2b6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8f7672e52f76877941f580--codecademy-styleguide.netlify.com>https://5e8f7672e52f76877941f580--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8f7672e52f76877941f580>Deploy Logs</a></p>
